app/src/api/usersApi.ts

```diff
@@ -28,35 +31,95 @@ export type User = {
  * @returns List of users
  */
 export async function fetchUsers(startIndex: number, length: number): Promise<User[]> {
-    try {
-        const [usersResponse, overviewText] = await Promise.all([
-            fetch(`${dataUrl.users}?skip=${startIndex}&limit=${length}`),
-            fetch(dataUrl.overview).then(res => res.text())
-        ]);
-
-        if (!usersResponse.ok) {
-            throw new Error(
-                `HTTP error! status: ${usersResponse.status}, statusText: ${usersResponse.statusText}`);
-        }
-
-        const data = await usersResponse.json();
-
-        return data.users.map((user: any) => ({
-            id: user.id,
-            fullName: `${user.firstName} ${user.lastName}`,
-            age: user.age,
-            gender: user.gender,
-            city: user.address.city,
-            stateCode: user.address.stateCode,
-            country: user.address.country,
-            email: user.email,
-            avatarUrl: `${dataUrl.avatar}${user.id % 70}`,
-            backgroundImageUrl: `${dataUrl.backgroundImage}${user.id}/200/300`,
-            overview: overviewText
-        }));
-
-    } catch (error) {
-        console.error('Failed to fetch users:', error);
-        throw error;
+    const [usersData, overviewText] = await Promise.all([
+        fetchData(`${dataUrl.users}?skip=${startIndex}&limit=${length}`),
+        fetchText(dataUrl.overview)
+    ]);
+
+    return usersData.users.map((user: any) =>
+        transformUserData(user, overviewText)
+    );
+}
+
+/**
+ * Fetches a single user by ID
+ * @param userId User ID
+ * @returns User object
+ */
+export async function fetchUser(userId: number): Promise<User> {
+    const [userData, overviewText] = await Promise.all([
+        fetchData(`${dataUrl.users}/${userId}`),
+        fetchText(dataUrl.overview)
+    ]);
+
+    return transformUserData(userData, overviewText);
+}
+
+/**
+ * Fetches a list of posts
+ * @param numPosts Number of posts to fetch
+ * @returns List of posts
+ */
+export async function fetchPosts(numPosts: number): Promise<PostProps[]> {
```

Comment:
Since this is just a toy example, it can remain as it is. However, normally, this API would be incorrect. You should always design APIs that support pagination, similar to the one we created here: `fetchUsers(startIndex: number, length: number)`.